apis/v1beta1/gateway_types.go

```diff
@@ -578,76 +595,65 @@ const (
 	// interoperability.
 	GatewayConditionAccepted GatewayConditionType = "Accepted"
 
-	// Deprecated: use "Accepted" instead.
-	GatewayConditionScheduled GatewayConditionType = "Scheduled"
-
 	// This reason is used with the "Accepted" condition when the condition is
 	// True.
 	GatewayReasonAccepted GatewayConditionReason = "Accepted"
 
+	// This reason is used with the "Accepted" condition when one or
+	// more Listeners have an invalid or unsupported configuration
+	// and cannot be configured on the Gateway.
+	// This can be the reason when "Accepted" is "True" or "False", depending on whether
+	// the listener being invalid causes the entire Gateway to not be accepted.
+	GatewayReasonListenersNotValid GatewayConditionReason = "ListenersNotValid"
+
+	// This reason is used with the "Accepted" and "Programmed"
+	// conditions when the status is "Unknown" and no controller has reconciled
+	// the Gateway.
+	GatewayReasonPending GatewayConditionReason = "Pending"
+
+	// This reason is used with the "Accepted" condition when the Gateway could not be configured
+	// because the requested address is not supported. This reason could be used in a number of
+	// instances, including:
+	//
+	// * The address is already in use.
+	// * The type of address is not supported by the implementation.
+	GatewaReasonUnsupportedAddress GatewayConditionReason = "UnsupportedAddress"
+)
+
+const (
+	// Deprecated: use "Accepted" instead.
+	GatewayConditionScheduled GatewayConditionType = "Scheduled"
+
 	// This reason is used with the "Scheduled" condition when the condition is
 	// True.
 	//
 	// Deprecated: use the "Accepted" condition with reason "Accepted" instead.
 	GatewayReasonScheduled GatewayConditionReason = "Scheduled"
 
-	// This reason is used with the "Accepted", "Programmed" and "Ready"
-	// conditions when the status is "Unknown" and no controller has reconciled
-	// the Gateway.
-	GatewayReasonPending GatewayConditionReason = "Pending"
-
 	// Deprecated: Use "Pending" instead.
 	GatewayReasonNotReconciled GatewayConditionReason = "NotReconciled"
-
-	// This reason is used with the "Accepted" condition when the
-	// Gateway is not scheduled because insufficient infrastructure
-	// resources are available.
-	GatewayReasonNoResources GatewayConditionReason = "NoResources"
 )
 
 const (
-	// Ready is an optional Condition that has Extended support. When it's set,
-	// the condition indicates whether the Gateway has been completely configured
-	// and traffic is ready to flow through the data plane immediately.
+	// "Ready" is a reserved condition type for future use. It should not be used by implementations.
+	// Note: This condition is not really "deprecated", but rather "reserved"; however, deprecated triggers Go linters
+	// to alert about usage.
```

Comment:
Suggest doesn't want to let move it, but placing this directly above the `Deprecated: Ready is reserved for future use` line may make more sense. I was a bit confused reading it here because there's no earlier mention of deprecation

Reply:
I touched-up some of the comments as part of https://github.com/kubernetes-sigs/gateway-api/pull/1934. I included this as well since it made more sense for it to be directly above the field. Not sure how other folks feel here.